### ISSUE TITLE
Deactivate aloha editor editing (AE 2.x)

### ISSUE THREAD

**Issue by Jotschi** [alohaeditor/Aloha-Editor#1301]
Add something like the old mahalo() to deactivate editing feature on an element.

**Comment by shannmcnicholl** [comment]
Any update on this issue? There seems to be a lack of input checking on the Undo.close method.

**Comment by Jotschi** [comment]
Did you reproduce the problem with the latest version of AE 2.x ?

**Comment by shannmcnicholl** [comment]
@Jotschi the version I'm testing with is being pulled from here: `http://aloha-editor.org/download/aloha.min.js`... if that answers your question... ?

**Comment by Jotschi** [comment]
Yes, thanks. I think @petrosalema is working on that issue.

**Comment by petrosalema** [comment]
I believe this commit https://github.com/alohaeditor/Aloha-Editor/commit/2cdc7a4f1d775c59e2e4ddfaa771d6c54a69147c addresses this issue.

**Comment by alfechner** [comment]
Why is this issue closed? I experience the exactly same problem and the fix in the commit quoted by @petrosalema seems not to be merged into productive branch, right?! I'm also testing with ````http://aloha-editor.org/download/aloha.min.js```` . Shouldn't the issue be open till the fix is included also in this version?

**Comment by petrosalema** [comment]
@alfechner, you're right the issue still happens with the minified version of aloha (aloha.min.js).
This appears to be an issue arising from Google Closure Compiler's advanced optimization munging an API point in the UndoContext objects.

**Comment by toovy** [comment]
@petrosalema are there any workarounds? I'm testing aloha with ember.js. Ember.js re-renders the HTML from if underlying data changes. I believe that I should properly deinitialize aloha before the re-render and the inintialize aloha again when the re-rendering is done. But it still gives me errors.

**Comment by Jotschi** [comment]
@toovy Have you tried the non minified version? That worked for me.

**Comment by toovy** [comment]
@Jotschi tried it now, thanks for the tip. Another error: 

    Uncaught TypeError: Cannot read property 'blinking' of null

The error is triggered in mahalo, first line, selections seems to be null:

    function mahalo(element) {
		editor.selection.blinking.stop(); // <- error
		Selections.hideCarets(element.ownerDocument);
		return Editables.destroy(editor, element);
	}

If I wrap the code in a simple if-statement it works:

    if (editor.selection) {
		  editor.selection.blinking.stop();
    }

Any thoughts?

Thanks, BR, toovy

**Comment by Jotschi** [comment]
@toovy When did you invoke mahalo? Did you activate any editable? Was the cursor visible? 
I assume it is still a valid case that mahalo should work even when no editable has been activated.

**Comment by toovy** [comment]
@Jotschi this could be related to #1360 , I've got strange issues with the carets. It happens no matter if I click the ediables or not (I guess that is meant with activate). As I'm using the same versions as the website uses I guess there could be some issues with other JS libs like jquery?
